src/ripple/app/ledger/Ledger.cpp

```diff
@@ -334,7 +334,7 @@ bool Ledger::addSLE (SLE const& sle)
 bool Ledger::addTransaction (uint256 const& txID, const Serializer& txn)
 {
     // low-level - just add to table
-    auto item = std::make_shared<SHAMapItem> (txID, txn.peekData ());
+    auto item = std::make_shared<SHAMapItem const> (txID, txn.peekData ());
```

Comment:
`TaggedCache` is a place to start looking, because `TreeNodeCache` is an alias for `TaggedCache`. I believe that as long as there is at least one outstanding `shared_ptr` to the `SHAMapItem` then things will be safe.